### PR TITLE
Add round_robin scheduling algorithm

### PR DESCRIPTION
**Pull request by pawanbuddy2000**
### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md  No type hints, no doctests, please format code with psf/black, please use snake_case_for_function_and_variable_names, etc.
https://travis-ci.com/github/TheAlgorithms/Python/builds/173491468#L252

**Comment by pawanbuddy2000** [comment]
Ok sir, I'll fix it thanks for the help

File: scheduling/round_robin_scheduling_algorithm.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+https://en.wikipedia.org/wiki/Round-robin_scheduling
+"""
+
+
+def calculate_waiting_time(processes, n, burst_time, waiting_time, quantum):
+    """
+    This function calculates the waiting time of some processes that have a
+    specified duration time.
+        Return: The waiting time for each process.
+    >>> calculate_waiting_times([5, 10, 15])
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:13]:
```suggestion
    >>> calculate_waiting_time([5, 10, 15])
```

File: scheduling/round_robin_scheduling_algorithm.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    [0, 5, 15]
+    >>> calculate_waiting_times([1, 2, 3, 4, 5])
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:15]:
```suggestion
    >>> calculate_waiting_time([1, 2, 3, 4, 5])
```

File: scheduling/round_robin_scheduling_algorithm.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    [0, 1, 3, 6, 10]
+    >>> calculate_waiting_times([10, 3])
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:17]:
```suggestion
    >>> calculate_waiting_time([10, 3])
```

File: scheduling/round_robin_scheduling_algorithm.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
+"""
+Round Robin is a CPU scheduling algorithm.
+In Round Robin each process is assigned a fixed time slot in a cyclic way.
+https://en.wikipedia.org/wiki/Round-robin_scheduling
+"""
+
+
+def calculate_waiting_time(processes, n, burst_time, waiting_time, quantum):
+    """
+    This function calculates the waiting time of some processes that have a
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:10]:
```suggestion
    Calculate the waiting time of some processes that have a
```

File: scheduling/round_robin_scheduling_algorithm.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    specified duration time.
+        Return: The waiting time for each process.
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:12]: This function should end with a Python `return` statement that provides the caller the waiting times...

`waiting_times = calculate_waiting_times()`

File: scheduling/round_robin_scheduling_algorithm.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            break
+
+
+def calculate_turn_around_time(
+    processes, n, burst_time, waiting_time, turn_around_time
+):
+
+    """
+    This function calculates the turnaround time of some processes.
+        Return: The time difference between the completion time and the
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:46]: This function should end with a Python `return` statement that provides the caller the turnaround times...

`turnaround_times = calculate_turnaround_times()`

File: scheduling/round_robin_scheduling_algorithm.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+                arrival time.
+                Practically waiting_time + duration_time
+    >>> calculate_turnaround_times([5, 10, 15], [0, 5, 15])
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:49]:
```suggestion
    >>> calculate_turnaround_time([5, 10, 15], [0, 5, 15])
```

File: scheduling/round_robin_scheduling_algorithm.py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    [5, 15, 30]
+    >>> calculate_turnaround_times([1, 2, 3, 4, 5], [0, 1, 3, 6, 10])
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:51]:
```suggestion
    >>> calculate_turnaround_time([1, 2, 3, 4, 5], [0, 1, 3, 6, 10])
```

File: scheduling/round_robin_scheduling_algorithm.py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    [1, 3, 6, 10, 15]
+    >>> calculate_turnaround_times([10, 3], [0, 10])
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:53]:
```suggestion
    >>> calculate_turnaround_time([10, 3], [0, 10])
```

File: scheduling/round_robin_scheduling_algorithm.py [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        if done is True:
+            break
+
+
+def calculate_turn_around_time(
+    processes, n, burst_time, waiting_time, turn_around_time
+):
+
+    """
+    This function calculates the turnaround time of some processes.
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:45]:
```suggestion
    Calculate the turnaround time of some processes.
```

File: scheduling/round_robin_scheduling_algorithm.py [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    [1, 3, 6, 10, 15]
+    >>> calculate_turnaround_times([10, 3], [0, 10])
+    [10, 13]
+    """
+
+    for i in range(n):
+        turn_around_time[i] = burst_time[i] + waiting_time[i]
+
+
+def calculate_avg_time(processes, n, burst_time, quantum):
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:61]:
```suggestion
def calculate_avg_time(processes: int, n: int, burst_time: int, quantum: int) -> float:
```
Functions should have Python type hints https://docs.python.org/3/library/typing.html

File: scheduling/round_robin_scheduling_algorithm.py [hunk 12]
```diff
@@ -0,0 +1,102 @@
+"""
```
Comment (cclauss) [inline scheduling/round_robin_scheduling_algorithm.py:1]: Let’s just call the file `round_robin.py`.

Comment (pawanbuddy2000) [reply scheduling/round_robin_scheduling_algorithm.py]: Thanks for the help sir...

**Comment by pawanbuddy2000** [comment]
Sir, that code wored....

**Comment by pawanbuddy2000** [comment]
Sir, my last code passes all the tests right....

**Comment by pawanbuddy2000** [comment]
Sir, Python type hints should be only be declared in List?

**Comment by cclauss** [comment]
> Python type hints functions should be only be declared in List?

Each of these functions is taking in a `list of ints` which is written in Python type hints as `List[int]`.

**Pull request merged**